tfs/src/main/java/hudson/plugins/tfs/TeamFoundationServerScm.java

```diff
@@ -400,17 +419,17 @@ public boolean pollChanges(AbstractProject hudsonProject, Launcher launcher, Fil
     }
     
     @Override
-    public boolean processWorkspaceBeforeDeletion(AbstractProject<?, ?> project, FilePath workspace, Node node) throws IOException, InterruptedException {
-        AbstractBuild<?, ?> lastRun = project.getLastBuild();
+    public boolean processWorkspaceBeforeDeletion(Job<?, ?> project, FilePath workspace, Node node) throws IOException, InterruptedException {
+        Run<?, ?> lastRun = project.getLastBuild();
         if (lastRun == null) {
             return true;
         }
         
         // Due to an error in Hudson core (pre 1.321), null was sent in for all invocations of this method
-        // Therefore we try to work around the problem, and see if its only built on one node or not. 
-        if (node == null) { 
+        // Therefore we try to work around the problem, and see if its only built on one node or not.
+        if (node == null) {
             while (lastRun != null) {
-                AbstractBuild<?,?> build = lastRun;
+                AbstractBuild<?,?> build = (AbstractBuild)lastRun;

```

Comment:
Should we check the "instanceOf" lastRun before casting here?

Reply:
From the comment I understood that node was only going to be NULL in Hudson core < 1.321, which doesn't support pipelines (so no Run class, just AbstractBuild). But I get that's a bit thin....I went ahead and added a check, just in case